### DISCUSSION TITLE
weighted-search

### DISCUSSION THREAD

**Discussion by MwSpaceLLC** [prisma/prisma#16222] [Q&A] [answered]
Hi there, today I'm going to create with my teame query for order by case as follow:

```sql
SELECT ragione_sociale, localita, nazione, macrocategoria, microcategoria,
IF(
`ragione_sociale` LIKE "bar%",  20,
IF(`ragione_sociale` LIKE "%bar%", 10, 0)
)
+ IF(`macrocategoria` LIKE "%bar%", 5,  0)
+ IF(`microcategoria`         LIKE "%bar%", 1,  0)
AS `weight`
FROM anagrafiche
WHERE (
    `ragione_sociale` LIKE "%bar%"
    OR `macrocategoria` LIKE "%bar%"
    OR `microcategoria`         LIKE "%bar%"
)
ORDER BY weight DESC
```

OR:

```sql
SELECT country_code,ragione_sociale, localita, nazione, macrocategoria, microcategoria
FROM anagrafiche
WHERE (
              `ragione_sociale` LIKE "%bar%"
              OR `macrocategoria` LIKE "%bar%"
              OR `microcategoria` LIKE "%bar%"
          )
ORDER BY CASE
             WHEN `ragione_sociale` LIKE "% bar %" THEN ragione_sociale
             WHEN `macrocategoria` LIKE "%bar%" THEN macrocategoria
             WHEN `microcategoria` LIKE "%bar%" THEN microcategoria
             END DESC
```

OR:

```sql
SELECT ragione_sociale,
       localita,
       nazione,
       macrocategoria,
       microcategoria,
       IF(MATCH (ragione_sociale) AGAINST ('bar' IN NATURAL LANGUAGE MODE ), 20, 0)
           + IF(MATCH (macrocategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE), 5, 0)
           + IF(MATCH (microcategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE), 1, 0)
           AS `weight`
FROM anagrafiche
WHERE country_code = 'IT'
  AND (
        MATCH (ragione_sociale) AGAINST ('bar' IN NATURAL LANGUAGE MODE)
        OR MATCH (macrocategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE)
        OR MATCH (microcategoria) AGAINST ('bar' IN NATURAL LANGUAGE MODE)
    )
ORDER BY weight DESC LIMIT 100
```

 We want use prisma modelling but we can't find the correct method in the guide.

The aspected records order depends by result in query (match like fulltextindex)

Tnks for any reply 🤖

**Comment by pantharshit00** [comment, marked as answer]
Hello, 

I don't think Prisma Client supports this SQL feature yet. You will need to use a raw query here: https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access

Also, can you please report this as a feature request so that we can also track support for this in the client. You can open it here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=

**Comment by MwSpaceLLC** [comment]
Mongodb support this:

https://www.mongodb.com/docs/atlas/atlas-search/tutorial/compound-query-custom-score/

**Reply by nurul3101** [reply]
You would need to use MongoDB Raw [Queries](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#runcommandraw) to achieve this, as weighted search is not natively supported in Prisma yet. I would also recommend creating a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for it, if you would like us to support it natively.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
